lib/block-supports/dimensions.php

```diff
@@ -74,6 +74,65 @@ function gutenberg_apply_dimensions_support( $block_type, $block_attributes ) {
 	return $attributes;
 }
 
+/**
+ * Renders server-side dimensions styles to the block wrapper.
+ * This block support uses the `render_block` hook to ensure that
+ * it is also applied to non-server-rendered blocks.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_dimensions_support( $block_content, $block ) {
+	$block_type               = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$block_attributes         = ( isset( $block['attrs'] ) && is_array( $block['attrs'] ) ) ? $block['attrs'] : array();
+	$has_aspect_ratio_support = block_has_support( $block_type, array( 'dimensions', 'aspectRatio' ), false );
+
+	if (
+		! $has_aspect_ratio_support ||
+		wp_should_skip_block_supports_serialization( $block_type, 'dimensions', 'aspectRatio' )
+	) {
+		return $block_content;
+	}
+
+	$dimensions_block_styles                = array();
+	$dimensions_block_styles['aspectRatio'] = $block_attributes['style']['dimensions']['aspectRatio'] ?? null;
+
+	$styles = gutenberg_style_engine_get_styles( array( 'dimensions' => $dimensions_block_styles ) );
+
+	if ( ! empty( $styles['css'] ) ) {
+		// Inject dimensions styles to the first element, presuming it's the wrapper, if it exists.
+		$tags = new WP_HTML_Tag_Processor( $block_content );
+
+		if ( $tags->next_tag() ) {
+			$existing_style = $tags->get_attribute( 'style' );
+			$updated_style  = '';
+
+			if ( ! empty( $existing_style ) ) {
+				$updated_style = $existing_style;
+				if ( ! str_ends_with( $existing_style, ';' ) ) {
+					$updated_style .= ';';
+				}
+			}
+
+			$updated_style .= $styles['css'];
+			$tags->set_attribute( 'style', $updated_style );
+
+			if ( ! empty( $styles['classnames'] ) ) {
+				foreach ( explode( ' ', $styles['classnames'] ) as $class_name ) {
+					$tags->add_class( $class_name );
```

Comment:
Mainly interested... so `$tags->add_class( 'a-class b-class' ) should work too?

Reply:
At the moment I'm calling `explode` so that we add them one-by-one... I looked up the `add_class` method and it does look like we should be able to call it with a full classname string that contains multiple classes, I just wasn't sure if it's _guaranteed_ to 🤔. Happy to switch it over to the simpler `add_class` with the full single string, though 🙂

Thanks for giving this PR an early look! 🙇

Reply:
I gave it a try and it works 😄 No biggie, I was just wondering.
<img width="650" alt="Screenshot 2023-12-12 at 4 43 30 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/b3001712-9ac2-45e1-beb0-1b147e9a37f9">

Reply:
Nice one! I can switch it over 😀

Reply:
Yeah if it's not explicitly indicated in the docs it might not be officially supported even though it currently works. We were using similar logic to fetch a bunch of classes from an element in one go in the layout support, but [this change](https://github.com/WordPress/wordpress-develop/pull/5096) broke that ability. The refactor of layout in #54075 to fix that has more info.

Reply:
Agree - might be a nice feature though to have an `add_classes` method that accepts an array or something.

Reply:
Good points! Related to what you've linked to in #54075, I just did a quick test of what happens if the class string includes multiple classes, and it seems it gets treated as a single classname so the de-duping logic in `add_class` gets skipped, resulting in potential duplicates:

<img width="667" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/0a763e53-a53d-4884-84a1-bcf5ccfe188b">

I'll stick with the `foreach` for now 🙂